packages/shorebird_cli/lib/src/commands/patch/windows_patcher.dart

```diff
@@ -0,0 +1,160 @@
+import 'dart:io';
+
+import 'package:crypto/crypto.dart';
+import 'package:mason_logger/mason_logger.dart';
+import 'package:path/path.dart' as p;
+import 'package:platform/platform.dart';
+import 'package:shorebird_cli/src/archive/archive.dart';
+import 'package:shorebird_cli/src/archive_analysis/windows_archive_differ.dart';
+import 'package:shorebird_cli/src/artifact_builder.dart';
+import 'package:shorebird_cli/src/artifact_manager.dart';
+import 'package:shorebird_cli/src/code_push_client_wrapper.dart';
+import 'package:shorebird_cli/src/commands/patch/patcher.dart';
+import 'package:shorebird_cli/src/doctor.dart';
+import 'package:shorebird_cli/src/executables/executables.dart';
+import 'package:shorebird_cli/src/logging/logging.dart';
+import 'package:shorebird_cli/src/patch_diff_checker.dart';
+import 'package:shorebird_cli/src/platform/platform.dart';
+import 'package:shorebird_cli/src/release_type.dart';
+import 'package:shorebird_cli/src/shorebird_flutter.dart';
+import 'package:shorebird_cli/src/shorebird_validator.dart';
+import 'package:shorebird_cli/src/third_party/flutter_tools/lib/flutter_tools.dart';
+import 'package:shorebird_code_push_client/shorebird_code_push_client.dart';
+
+/// {@template windows_patcher}
+/// Functions to create a Windows patch.
+/// {@endtemplate}
+class WindowsPatcher extends Patcher {
+  /// {@macro windows_patcher}
+  WindowsPatcher({
+    required super.argParser,
+    required super.argResults,
+    required super.flavor,
+    required super.target,
+  });
+
+  @override
+  // TODO(bryanoltman): exe isn't technically correct - we upload a zip
```

Comment:
Will this be a pain to change later?

Reply:
We would probably want to change it before launching

Reply:
Changed to `win_archive`